front/dashboard/src/app/components/filtro/filtro.component.ts

```diff
@@ -12,8 +12,29 @@ export class FiltroComponent {
   tipos_documento: any
   buttonsDoc: any
   buttonsEmp: any
+  segmentos: any
 
+  async Filterseg(){
+    let res = await getUsers();
+    this.empresas = res.data.user;
```

Comment:
Se passar a chamada de Filterseg() pra depois da linha 39 não precisaria fazer essa requisição porque this.empresas já vai estar com as empresas